core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -293,6 +296,24 @@ public HttpResponse doCreateAdminUser(StaplerRequest req, StaplerResponse rsp) t
             }
         }
     }
+    
+    @RequirePOST
+    public HttpResponse doConfigureRootUrl(StaplerRequest req, @QueryParameter String rootUrl) {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+        
+        LOGGER.log(Level.FINE, "Root URL set during SetupWizard to {0}", new Object[]{ rootUrl });
+        
+        JenkinsLocationConfiguration.getOrDie().setUrl(rootUrl);
```

Comment:
nit: Could be nice to try parsing the URL and return an error if it isn't a valid URL. (You could also do the same thing in the admin monitor and have the message state that the URL appears to be invalid.)